backend/src/Edna/Upload/DB/Query.hs

```diff
@@ -26,21 +27,36 @@ import Edna.Util
   (CompoundId, ExperimentFileId, ExperimentId, MeasurementId, MethodologyId, ProjectId, SqlId(..),
   SubExperimentId, TargetId, fromSqlSerial)
 
--- | Insert experiment and return its ID
-insertExperiment :: ExperimentFileId -> CompoundId -> TargetId -> Edna ExperimentId
-insertExperiment (SqlId experimentFileId) (SqlId compoundId) (SqlId targetId) =
-  fromSqlSerial . eExperimentId <$>
-    runInsertReturningOne' (insert (esExperiment ednaSchema) $ insertExpressions
-      [ ExperimentRec
+-- | Insert experiments and return its IDs
+insertExperiments :: [(ExperimentFileId, CompoundId, TargetId)] -> Edna [ExperimentId]
+insertExperiments experiments = map (fromSqlSerial . eExperimentId) .
+  sortWith (\a -> (eCompoundId a, eTargetId a)) <$>
+    runInsertReturningList' (insert (esExperiment ednaSchema) $ insertExpressions $
+      flip map experiments $ \(SqlId experimentFileId, SqlId compoundId, SqlId targetId) ->
+        ExperimentRec
         { eExperimentId = default_
         , eExperimentFileId = val_ experimentFileId
         , eCompoundId = val_ compoundId
         , eTargetId = val_ targetId
-        }
-      ])
+        })
+
+-- | Insert sub-experiments and return its IDs
+insertSubExperiments :: [Params4PLResp] -> Edna [SubExperimentId]
+insertSubExperiments params = map (fromSqlSerial . seSubExperimentId) .
+  sortBy (\ a b -> seExperimentId a `compare` seExperimentId b) <$>
```

Comment:
There are more places (this one + 2) where `sortBy` is used with `compare` inside and apparently can be simplified using `sortWith` or `sortOn`.